server/api/hook.go

```diff
@@ -237,6 +237,16 @@ func PostHook(c *gin.Context) {
 	}
 
 	if build.Status == model.StatusBlocked {
+		// TODO: check of build.Procs contains proper data
```

Comment:
@6543 Could you help me with this part. I need to have `build.Procs` at this point to set the correct status. As this hook call is just creating the build from ground up, procs are not retrived from the pipeline config. That's done on approval or later if the repo is not protected / build is not blocked by `startBuild`. Should I just extract following code from `startBuild` to a function and call that function in `startBuild` and also at this todo?

```go
b := shared.ProcBuilder{
		Repo:  repo,
		Curr:  build,
		Last:  last,
		Netrc: netrc,
		Secs:  secs,
		Regs:  regs,
		Envs:  envs,
		Link:  server.Config.Server.Host,
		Yamls: yamls,
	}
	buildItems, err := b.Build()
	if err != nil {
		if _, err := shared.UpdateToStatusError(store, *build, err); err != nil {
			log.Error().Err(err).Msgf("Error setting error status of build for %s#%d", repo.FullName, build.Number)
		}
		return nil, err
	}
	build = shared.SetBuildStepsOnBuild(b.Curr, buildItems)
```

Reply:
to be fair, I would test how it behaves and if we can move that section :point_up: above `build.Status == model.StatusBlocked` ... we should do it

Reply:
@6543 `build.Procs` was empty at this place, but I fixed it by spiting your function `startBuild` into `createBuild` and `startBuild`. I already tested the code a bit and it seems to work quite nicely already. By splitting the two functions I also deduped the code of `PostBuild` which is used to restarts build a bit.